main/src/main/scala/sbt/Main.scala

```diff
@@ -672,7 +673,26 @@ object BuiltinCommands {
   def loadProjectImpl: Command =
     Command(LoadProjectImpl)(_ => Project.loadActionParser)(doLoadProject)
 
+  def checkSBTVersionChanged(state: State): Unit = {
+    import sbt.io.syntax._
+    val app = state.configuration.provider
+    val buildProps = state.baseDir / "project" / "build.properties"
+    // First try reading the sbt version from build.properties file.
+    val sbtVersionOpt = if (buildProps.exists) {
+      val buildProperties = new Properties()
+      IO.load(buildProperties, buildProps)
+      Option(buildProperties.getProperty("sbt.version"))
+    } else None
+
+    sbtVersionOpt.foreach(version =>
+      if (version != app.id.version()) {
+        state.log.warn(
+          "sbt version in build.properties file has changed, use 'reboot' instead of 'reload' to use the new value")
```

Comment:
I think it would be better to include detail on how it changed. So maybe:

```scala
s"""sbt version mismatch, current: ${app.id.version()}, in build.properties: "$version", use 'reboot' to use the new value."""
```

Reply:
I agree it looks nicer. I'm concerned this would make scripted test a bit tricky, because current version will be constantly changing as the sbt evolves. One "easy" way is, obviously, hardcode the current sbt version number in the scripted test and update it every time the sbt version is bumped, but it has obvious disadvantages. Is there a better way?

Reply:
You can still access `state.value.configuration.provider.id.version()` in the scripted test.

If that fails just search-replace that part of the message to ignore it.

Reply:
Right, that works. Pushed the change.